### PR TITLE
Add Imprint and Privacy Policy into mails

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
- Add reusable mail html and plaintext footers
- Add the links in footers if needed

## Related Issue
https://github.com/owncloud/enterprise/issues/2537

## Motivation and Context
gdpr

## How Has This Been Tested?
Manually

## Screenshots (if appropriate):
Plaintext
plaintext

Html
html

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: core/templates/plain.mail.footer.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--
+<?php p($theme->getName() . ' - ' . $theme->getSlogan()); ?>
+<?php
+print_unescaped("\n" . $theme->getBaseUrl());
+ if ($theme->getImprintUrl() !== '') {
```
Comment (PVince81) [inline core/templates/plain.mail.footer.php]: a wild sneaky space appeared

File: core/templates/html.mail.footer.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+--<br>
+<?php p($theme->getName()); ?> -
+<?php p($theme->getSlogan()); ?>
+<br><a href="<?php p($theme->getBaseUrl()); ?>"><?php p($theme->getBaseUrl());?></a>
+<?php if ($theme->getImprintUrl() !== ''): ?>
+<br><a href="<?php print_unescaped($theme->getImprintUrl()); ?>"><?php p($l->t('Imprint'));?></a>
+<?php endif; ?>
+<?php if ($theme->getPrivacyPolicyUrl() !== ''): ?>
+	<br><a href="<?php print_unescaped($theme->getPrivacyPolicyUrl()); ?>"><?php p($l->t('Privacy Policy'));?></a>
```
Comment (PVince81) [inline core/templates/html.mail.footer.php]: should we escape the URL in all href using `p()` ? see above how the getBaseUrl is also escaped

**Review by PVince81 — DISMISSED** [review]
Some concerns about unescaped URLs

**Comment by PVince81** [comment]
@VicDeo I found this PR https://github.com/owncloud/core/pull/31144 that was actually ready to merge so I merged it. Please also adjust it after rebase if you didn't adjust yet.

- [ ] adjust/readjust lost password emails

File: core/templates/html.mail.footer.php (hunk 2 above)
Comment (PVince81) [reply core/templates/html.mail.footer.php]: @VicDeo please escape all urls

File: core/templates/html.mail.footer.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+--<br>
```
Comment (PVince81) [inline core/templates/html.mail.footer.php]: assuming these "--" is expected ? strangely in the red part above it was three dashes instead of two

Comment (PVince81) [reply core/templates/html.mail.footer.php]: >  the first dash is github-specific and means that line was removed :)

face... palm...

**Comment by VicDeo** [comment]
@PVince81 replaced `print_unescaped` with `c`

**Review by PVince81 — APPROVED** [review]
:+1:

File: lib/private/legacy/defaults.php [hunk 4]
```diff
@@ -46,10 +46,14 @@ class OC_Defaults {
 	private $defaultDocVersion;
 	private $defaultSlogan;
 	private $defaultLogoClaim;
-	private $defaultMailHeaderColor;
```
Comment (VicDeo) [inline lib/private/legacy/defaults.php]: @PVince81 I noticed that this line was deleted accidentally while doing a backport :cry:

Comment (PVince81) [reply lib/private/legacy/defaults.php]: looking forward to the PR that adds it back

**Comment by phil-davis** [comment]
Backport ``stable10`` #31699

**Pull request merged**